### PR TITLE
updating UI of Navbar

### PR DESCRIPTION
**Pull request by Lobi29**
<h1 align="center">ISSUE #278</h1>

### Feature Added

- "Contribute Now" option link
- Navbar made sticky with backdrop blur effect
- Direct links to leaderboard, contributors, feed, and projects
- Full Responsive

### Mobile View:

Screenshot from 2024-02-29 10-08-13

Screenshot from 2024-02-29 10-08-05

### Desktop View

Screenshot from 2024-02-29 10-07-11
Screenshot from 2024-02-29 10-06-58

### CODE REVIEW

**Comment by Lobi29** [comment]
### changing the gradient to a more asthetic and minimal one

Untitled drawing

- if it's all right, then please merge it 😊

File: components/navbar/ContributeButton.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+"use client";
```
Comment (rithviknishad) [inline components/navbar/ContributeButton.tsx:1]: Why is this a `use client` component?

File: components/navbar/ContributeButton.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import Link from "next/link";
+import { AiFillGithub } from "react-icons/ai";
+
+const ContributeButton = () => {
+  return (
+    <Link
+      href="https://github.com/coronasafe/leaderboard/issues"
```
Comment (rithviknishad) [inline components/navbar/ContributeButton.tsx:8]: Leaderboard would be deployed by different organizations, it'd be great to open the organization as the viewer might be looking to contribute to the organization.

Organization-specific data is not kept in this repo.
Use environment variables.

File: app/globals.css [hunk 3]
```diff
@@ -136,3 +136,9 @@ html.dark {
   transform: translateX(0px);
   transform: translateY(0px);
 }
+
+.text-gradient {
+  background: linear-gradient(45deg, #6366f1, #1d4ed8);
```
Comment (rithviknishad) [inline app/globals.css:141]: What if the organization deploying the leaderboard has a different colour theme? #283

File: components/navbar/Logo.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+"use client";
```
Comment (rithviknishad) [inline components/navbar/Logo.tsx:1]: Why use client again?

```suggestion
```

File: components/navbar/MenuItems.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
+"use client";
```
Comment (rithviknishad) [inline components/navbar/MenuItems.tsx:1]: Why use client?

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: components/navbar/ContributeButton.tsx (hunk 1 above)
Comment (Lobi29) [reply components/navbar/ContributeButton.tsx:1]: sorry for this silly mistake, I am removing this line from this component

File: components/navbar/ContributeButton.tsx (hunk 2 above)
Comment (Lobi29) [reply components/navbar/ContributeButton.tsx:8]: ok, I will update it with .env varible

File: app/globals.css (hunk 3 above)
Comment (Lobi29) [reply app/globals.css:141]: okay, I will use the colour theme from the tailwind.config.ts file

File: components/navbar/Logo.tsx (hunk 4 above)
Comment (Lobi29) [reply components/navbar/Logo.tsx:1]: again sorry

File: components/navbar/MenuItems.tsx (hunk 5 above)
Comment (Lobi29) [reply components/navbar/MenuItems.tsx:1]: again again sorry 😊

**Comment by Lobi29** [comment]
<h1 align="center">Updates</h1>

- Added `.env` variable to the contribution link's href
- Color theme applied, with emphasis on the primary theme
- 'use client' feature removed from redundant components

**Review by rithviknishad — CHANGES_REQUESTED** [review]
@Lobi29 

1. The gradient still feels like the text is not very readable. More contrast between the background is needed. Let's keep it as minimal and aesthetic as possible.

image


2. Let's keep the old sidebar in the mobile view itself. This is because in future we may have more links to be shown.

image

3. The theme switcher is getting more attention than required due to the strong contrast from the background. Let's keep the colors of the theme switcher like how it was before.

image

4. The navbar looked good before. 
image

**Comment by Lobi29** [comment]
Got it 😁👍,

**Comment by Lobi29** [comment]
color is updated

**Review by rithviknishad — CHANGES_REQUESTED** [review]
@Lobi29 

I don't think this sidebar would look good on mobile. Let's have the old sidebar that takes the full height of the screen itself and show the theme switcher inside the sidebar in mobile view. Let's not do blur for the slide over part of the sidebar as it'd make the text unreadable. (Fine with navbar).

image



Also, you've not made the text lighter yet to bring in more contrast from the background.
It is still not very readable.

image

image

**Review by rithviknishad — CHANGES_REQUESTED** [review]
Also use the default font configured (that is font-mono) across the leaderboard. Having font-inter in just some places of leaderboard doesn't make it look good.

**Comment by Lobi29** [comment]
Untitled drawing (4)

- Drawing from your previous advice, I've refined the sidebar and subtly adjusted the text's hue, enhancing its readability and aesthetic appeal

**Review by rithviknishad — COMMENTED** [review]
LGTM, just one last suggestion

image

**Comment by Lobi29** [comment]
sure, why not 😊

**Comment by Lobi29** [comment]
Hey @rithviknishad , "Contribute Now" text has been removed, leaving only the GitHub button visible.😊

**Review by rithviknishad — APPROVED** [review]

**Pull request merged**